src/imageio/JxlImageLoader.cpp

```diff
@@ -0,0 +1,233 @@
+                            // Set up conversion matrix - simplified approach
+                            // In a real implementation, we'd need to properly extract the primaries
+                            // from the ICC profile and calculate an accurate conversion matrix
+                            toRec709 = Matrix4f(1.0); // Placeholder
+                        }
+                    }
+                }
+            }
+            
+            status = JxlDecoderProcessInput(decoder.get());
+        } else if (status == JXL_DEC_NEED_IMAGE_OUT_BUFFER) {
+            size_t buffer_size;
+            if (JXL_DEC_SUCCESS != JxlDecoderImageOutBufferSize(decoder.get(), &format, &buffer_size)) {
+                throw runtime_error{"Failed to get output buffer size for JPEG XL image."};
+            }
+
+            pixels.resize(buffer_size / sizeof(float));
+            if (JXL_DEC_SUCCESS != JxlDecoderSetImageOutBuffer(decoder.get(), &format, pixels.data(), buffer_size)) {
+                throw runtime_error{"Failed to set output buffer for JPEG XL image."};
+            }
+
+            // Read the pixel data into the pixels buffer
+            status = JxlDecoderProcessInput(decoder.get());
+
+            // Check if we got any pixels
+            if (pixels.empty()) {
+                throw runtime_error{"Failed to decode JPEG XL image data."};
+            }
+
+            result.emplace_back();
+            ImageData& data = result.back();
+
+            // Set dimensions
+            Vector2i size{(int)info.xsize, (int)info.ysize};
+            data.dataWindow = {
+                {0,        0       },
+                {size.x(), size.y()}
+            };
+            data.displayWindow = data.dataWindow;
+            data.hasPremultipliedAlpha = false;
+
+            // Set color transform if needed
+            if (toRec709 != Matrix4f(1.0)) {
+                data.toRec709 = toRec709;
+            }
+
+            // Create channels
+            int numChannels = format.num_channels;
+            vector<Channel> channels = makeNChannels(numChannels, size);
+
+            // Check if the channel selector matches any of our channels
+            bool matchFound = false;
+            for (int i = 0; i < numChannels; ++i) {
+                if (matchesFuzzy(channels[i].name(), channelSelector)) {
+                    matchFound = true;
+                    break;
+                }
+            }
+
+            if (!matchFound && !channelSelector.empty()) {
+                throw invalid_argument{fmt::format("No channels match '{}'.", channelSelector)};
+            }
+
+            // Copy pixels to channels
+            co_await ThreadPool::global().parallelForAsync<int>(
+                0,
+                size.y(),
+                [&](int y) {
+                    for (int x = 0; x < size.x(); ++x) {
+                        size_t pixelIdx = ((size_t)y * size.x() + x) * numChannels; // RGBA format from JXL decoder
```

Comment:
This index should be, roughly,
```
size_t align = std::max(format.align, (size_t)1); // Handle case where format.align == 0
size_t bytesPerRow = size.x() * sizeof(float) * numChannels;
bytesPerRow = (bytesPerRow + align - 1) / align * align;
size_t pixelIdx = y * bytesPerRow / sizeof(float) + x * numChannels;
```

From `JxlPixelFormat`'s definition:
```
  /** Align scanlines to a multiple of align bytes, or 0 to require no
   * alignment at all (which has the same effect as value 1)
   */
  size_t align;
```

Reply:
Following up: I just realized you're the one in control of `format.align`, making the above comment moot. 😅

Reply:
It might still be useful for the other bit depths or doing some kind of bulk memcpy 😄

Speaking of other bit depths: Is there any scenario where tev would want the images in uint8 or uint16 if the underlying image reported itself that way?